Add documentation page for ACME Profiles

aarongable opened the pull request:
Add a new page describing what a profile is, what a profile controls, what profiles we have, and how to request a specific profile.

Link to this new page from the Documentation table of contents and from the Glossary.

Preston12321 commented:
I suggest the bulleted lists explaining validation and certificate properties be moved to the end of the doc, the "Selecting a Profile" section be moved up, and the tables for each profile be combined into a single table toward the top of the page.

While it's important the subscriber understands what each profile-controlled characteristic means and why they might choose a particular profile over another, I think the most important bits of information this page needs to convey are the concrete profile options available and how to select one via an ACME client. If I were trying to use the docs as a reference, these are the things I'd want to be able to read off the page most immediately.

I've made #1835 to show a rough outline of what I'm thinking.

aarongable commented:
Okay, I've uploaded a new version in response to comments:
- @beautifulentropy I'm now using the "boxed" style to set apart the three profiles
- @Preston12321 I've moved the profiles up to the top, moved the field definitions down to the bottom, and hyperlinked from the tables to the definitions. I have not combined the tables into a single table.

Let me know what you think!

Preston12321 commented on content/en/docs/profiles.md:
```diff
@@ -0,0 +1,132 @@
+
+## Validation Properties
+
+The following validation properties can be controlled by our profiles:
```
nit: Ending the line with a colon looks weird now that it's immediately followed by a second-level heading. Maybe something like this?

```suggestion
Below are descriptions of the validation properties that can be controlled by our profiles.
```

Preston12321 commented on content/en/docs/profiles.md:
```diff
@@ -0,0 +1,132 @@
+
+## Certificate Properties
+
+The following certificate properties can be controlled by our profiles:
```
nit: Ending the line with a colon looks weird now that it's immediately followed by a second-level heading. Maybe something like this?

```suggestion
Below are descriptions of the certificate properties that can be controlled by our profiles.
```

Preston12321 reviewed (commented):
Much happier with the layout now, thanks! The "boxed" style really helps visually indicate the three options, so I'm good with not having the master table at the top

aarongable commented:
This page is now ready for final review and publication. Please take a look.

inahga commented on content/en/docs/profiles.md:
```diff
@@ -0,0 +1,132 @@
+
+## tlsserver
+
+The tlsserver profile is a new profile which updates several of these validation and certificate properties in order to reflect the latest recommendations from the CA/Browser Forum Baseline Requirements, as well as general trends within the WebPKI community. We recommend selecting this profile for subscribers who want smaller certificates and who fully embrace automation.
```
Is it worth saying something like:
> general trends within the WebPKI community, **as of 2025**.

Being that there will be new recommendations that we might not necessarily roll into the tlsserver profile immediately. Or is it the intent that tlsserver will be continuously updated?

aarongable replied:
We do intend to evolve the tlsserver profile as requirements and best practices change.

inahga commented on content/en/docs/profiles.md:
```diff
@@ -0,0 +1,132 @@
+
+## shortlived
+
+The shortlived profile is identical to the tlsserver profile, with one key distinction: the resulting certificate is only valid for 6ish days. This allows these certificates to qualify as "Short-Lived Subscriber Certificates" under the Baseline Requirements, which means they do not need to contain any revocation information. This makes the certificates even smaller, and removes any possibility of ...
```
The awkward thing about this is that the short-lived certs we're issuing will contain the OCSP URI. So we don't _have_ to contain revocation information, but we do.

aarongable replied:
Yeah, I chose the phrasing "do not _need_ to contain" carefully. But I've rephrased slightly to be even more cagey about it.

inahga reviewed (approved):
LGTM!

aarongable commented:
Merging now. We can continue to edit and update this page after it goes live, if other folks have comments and suggestions.
